### ISSUE TITLE
CI builds fail on python 3.10 with fastcluster > 1.1.25

### ISSUE THREAD

**Issue by NickCrews** [dedupeio/dedupe#950]
Originally from found in https://github.com/dedupeio/dedupe/issues/949, when that CI run failed.

Looking at the [logs of a recent successful build](https://github.com/dedupeio/dedupe/runs/4970736852?check_suite_focus=true), looks like it's only a problem for python 3.10. Other versions of python install fastcluster 1.2.4 just fine. In that python 3.10 run, CI tried to build fastcluster 1.2.4, failed, then kept on downgrading versions of fastcluster until it finally succeeded with fastcluster 1.1.25.

Here is [the diff between 1.1.25 -> 1.1.26](https://github.com/dmuellner/fastcluster/compare/v1.1.25...v1.1.26), not sure yet what the actual problem is...

Edit: ~~Something to do how fastcluster uses `setup.cfg` after 1.1.25, and [distutils is deprecated in 3.10?](https://docs.python.org/3/whatsnew/3.10.html#distutils-deprecated)~~ Don't think this is it.

**Comment by NickCrews** [comment]
Oh, it's because the latest released version of fastcluster (1.2.4) and earlier don't list a numpy dependency for python 3.10. [That was only added quite recently](https://github.com/dmuellner/fastcluster/compare/v1.2.4...master) and it hasn't been released.

Until that is released, maybe skip python 3.10 in CI? MArk as optional? not sure

**Comment by fgregg** [comment]
I had increased the timeout time previously to deal with. I'll bump it up again.

**Comment by fgregg** [comment]
hm.. this is strange. there should be a pyproject.toml in the sdist, so why is it failing so hard.

**Comment by fgregg** [comment]
there is a pyproject.toml, so something must have changed in the recent release of pip that changed how it handled this.

**Comment by fgregg** [comment]
yes, but the pyproject toml looks like this

```
[build-system]
# We specify older version of numpy for the build environment because
# numpy tries to maintain backwards compatibility but not forward
# compatibility. So, if a user tries to use an extension that was compiled
# along with an earlier version of numpy (say 1.12) than the user has on
# their current system, that will work (say 1.15). However, if the
# extension was compiled with a numpy 1.15 and the user has 1.12 on their
# system, the extension will not work.
#
# For each version of Python, we choose the earliest versions of numpy that
# has binary wheels for that Python version for window, mac, and linux.
# Otherwise building fastcluster would require compiling numpy which
# would make things much, much slower.
requires = ["setuptools",
            "wheel",
            "numpy==1.10.4; python_version<'3.6'",
            "numpy==1.12.1; python_version=='3.6'",
            "numpy==1.15.0; python_version=='3.7'",
            "numpy==1.17.3; python_version=='3.8'",
            "numpy==1.19.3; python_version=='3.9'"]
```

notice that there is no fallback for if python_version is greater than 3.9

**Comment by fgregg** [comment]
opened up an issue on fastcluster https://github.com/dmuellner/fastcluster/issues/30

**Comment by NickCrews** [comment]
Yup, that looks like the source of the problem. Just need to wait for a new release of fastcluster, until then pin to the fixed version when using python 3.10
